micro-infra-camel/src/main/java/com/ofg/infrastructure/camel/CorrelationIdInterceptor.java

```diff
@@ -65,8 +65,12 @@ private Span getCorrelationId(Exchange exchange) {
             log.debug("No spanId has been set in request inbound message. Creating new one.");
             spanId = idGenerator.nextLong();
         }
-        return builder().spanId(spanId).traceId(firstNonNull(oldTraceId, traceId))
-                .name(notSampledName).parent(parentId).processId(processID).build();
+        Span.SpanBuilder spanBuilder = builder().spanId(spanId).traceId(firstNonNull(oldTraceId, traceId))
+                .name(notSampledName).processId(processID);
+        if (parentId != null) {
+            spanBuilder.parent(parentId);
```

Comment:
`SpanBuilder` itself doesn't guard against nulls here, so it justs adds `null` to `List<Long> parents` which results in `[null]`.
After that whenever `toString()` or `getParentIdIfPresent()` is called, there is NPE, because we have `idToHex(this.getParents().get(0))` - `this.getParents().get(0)` is `null` and `idToHex` method requires `long` so there us automatic auto-unboxing which throws NPE